### PR TITLE
Update User and Maintainer Docs

### PR DESCRIPTION
**Pull request by krithika369**
# Description

This PR updates the user docs and adds minimal maintainer docs for Merlin.

**Note:** Both the sets of docs are templatised using Jinja2 and the `generated/*` folder contains the final docs for Gitbook that are created by running the following locally:

```sh
make docs
```

# Modifications

* `docs/.gitignore` - Ignore the [mdformatter](https://github.com/caraml-dev/mdformatter) files used for formatting
* `docs/Makefile` - Add make targets for the docs
* `docs/README.md` - Include broad info about the different types of docs in the repo
* `docs/SUMMARY.md` - This file is not needed by the [CaraML Gitbook docs](https://docs.caraml.dev/introduction/readme) and is removed.
* `docs/developer/` - Minor corrections to the links in the dev docs
* `docs/images/` - Add / update images
* `docs/maintainer/` - Add maintainer docs.
    - `docs/maintainer/values.json` - Values file for the maintainer docs
    - A comment with the `page-title`  is added to the `.md` files, to be able to publish the docs to Confluence, if needed.
    - The `generated/` subfolder contains the final generated docs.
* `docs/user/` - Add user docs.
    - `docs/user/values.json` - Values file for the maintainer docs
    - A comment with the `page-title` and `parent-e-title`  is added to the `.md` files, to be able to publish the docs to Confluence, if needed.
    - The `generated/` subfolder contains the final generated docs.
* `examples/` - Renaming the notebooks referenced by the user docs to replace spaces with hyphens, to overcome limitation # 4 of the doc formatting tool: https://github.com/caraml-dev/mdformatter?tab=readme-ov-file#limitations

# Tests

The following updates / tests are to be done after this PR is merged:
- [ ] Update the `caraml-dev/docs` repo, in particular, the [SUMMARY.md](https://github.com/caraml-dev/docs/blob/main/SUMMARY.md) file, to point to the `generated/**` doc files from Merlin repo.
- [ ] Verify that the [CaraML docs](https://docs.caraml.dev/introduction/readme) are working as expected

# Checklist
- [x] Added PR label
- [ ] Added unit test, integration, and/or e2e tests
- [x] Tested locally
- [x] Updated documentation
- [ ] Update Swagger spec if the PR introduce API changes
- [ ] Regenerated Golang and Python client if the PR introduces API changes

# Release Notes

```release-note
Update User and Maintainer Docs
```

### CODE REVIEW

File: docs/user/connecting-to-merlin/merlin-cli.md [hunk 1]
```diff
@@ -1,28 +0,0 @@
-# Merlin CLI
```
Comment (krithika369) [inline docs/user/connecting-to-merlin/merlin-cli.md:1]: The Merlin CLI is not very commonly used and is thus removed from the docs.

File: docs/Makefile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	@rm -rf mdformatter
+	@git clone https://github.com/caraml-dev/mdformatter.git
+	@pip install -r mdformatter/requirements.txt
+
+# The target below uses a non-existent doc overrides folder name to generate the final docs,
+# as there are no overrides.
+.PHONY: format
+format:
+	@echo "Formatting maintainer docs ..."
+	@cd mdformatter && python -m mdformatter ../maintainer/templates ../maintainer/overrides ../maintainer/generated ../maintainer/values.json GITBOOK
```
Comment (leonlnj) [inline docs/Makefile:15]: `../maintainer/overrides` doesn't have to be physical there is not required as per this repo right?

**Review by leonlnj — APPROVED** [review]
Thanks LGTM, dont have any concern. 

Had some doubt about how the merging/override works but it is address in https://github.com/caraml-dev/mdformatter?tab=readme-ov-file#formatting 

One more minor clarification, so the workflow for updating docs 
1. Edit template/values
2. Make docs
3. Verify generated MD locally 
4. Commit and push generated MD to branch
5. MR

File: docs/Makefile (hunk 2 above)
Comment (krithika369) [reply docs/Makefile:15]: Correct, it doesn't need to be there physically. If the specified override directory is not found, the merger tool will ignore it.

**Comment by krithika369** [comment]
> Thanks LGTM, dont have any concern.
> 
> Had some doubt about how the merging/override works but it is address in https://github.com/caraml-dev/mdformatter?tab=readme-ov-file#formatting
> 
> One more minor clarification, so the workflow for updating docs
> 
> 1. Edit template/values
> 2. Make docs
> 3. Verify generated MD locally
> 4. Commit and push generated MD to branch
> 5. MR

Yes, that's correct.

**Comment by krithika369** [comment]
Thanks for the quick review, @leonlnj ! Will merge this PR shortly.

**Pull request merged**